### PR TITLE
GUACAMOLE-1733: Add missing optional configuration options from docker start.sh script

### PR DESCRIPTION
**Pull request by ronansalmon**
Add missing optional configuration options from docker start.sh script

### CODE REVIEW

File: guacamole-docker/bin/start.sh [hunk 1]
```diff
@@ -1164,6 +1174,16 @@ if [ -n "$API_SESSION_TIMEOUT" ]; then
...
 fi
 
+# Maximum number of bytes to accept within the entity body of any particular HTTP request
+set_optional_property "api-max-request-size" "$API_MAX_REQUEST_SIZE"
+
+# A comma-separated whitelist of language keys to allow as display language 
+# choices within the Guacamole interface
+set_optional_property "allowed-languages" "$ALLOWED_LANGUAGES"
+
+set_optional_property "enable-environment-properties" "$ENABLE_ENVIRONMENT_PROPERTIES"
```
Comment (necouchman) [inline guacamole-docker/bin/start.sh:1184]: This should probably have a comment associated with it.

**Review by necouchman — CHANGES_REQUESTED** [review]
Only one minor change from me.

File: guacamole-docker/bin/start.sh (hunk 1 above)
Comment (ronansalmon) [reply guacamole-docker/bin/start.sh:1187]: Missed that comment.
Fixed.

File: guacamole-docker/bin/start.sh [hunk 2]
```diff
@@ -1064,6 +1069,11 @@ fi
 set_property "guacd-hostname" "$GUACD_HOSTNAME"
 set_property "guacd-port"     "$GUACD_PORT"
 
+# A comma-separated list of the identifiers of authentication providers that
+# should be allowed to fail internally without aborting the authentication process
+set_property "skip-if-unavailable"     "$SKIP_IF_UNAVAILABLE"
```
Comment (necouchman) [inline guacamole-docker/bin/start.sh:1074]: Shouldn't this be `set_optional_property`? Using `set_property` will put an empty string in there or possibly other unexpected results if it isn't present.

**Review by necouchman — CHANGES_REQUESTED** [review]
One other change.

File: guacamole-docker/bin/start.sh (hunk 2 above)
Comment (ronansalmon) [reply guacamole-docker/bin/start.sh]: makes sense!

File: guacamole-docker/bin/start.sh [hunk 3]
```diff
@@ -1164,6 +1174,19 @@ if [ -n "$API_SESSION_TIMEOUT" ]; then
     associate_apisessiontimeout
 fi
 
+# Maximum number of bytes to accept within the entity body of any particular HTTP request
+set_optional_property "api-max-request-size" "$API_MAX_REQUEST_SIZE"
+
+# A comma-separated whitelist of language keys to allow as display language
```
Comment (necouchman) [inline guacamole-docker/bin/start.sh:1180]: I would suggest changing this to:
`A comma-separated list of language keys to allow...`

**Review by necouchman — CHANGES_REQUESTED** [review]
And one more minor wording change in a comment :-).

File: guacamole-docker/bin/start.sh (hunk 3 above)
Comment (ronansalmon) [reply guacamole-docker/bin/start.sh:1180]: ok

**Comment by necouchman** [comment]
Okay, the changes are good, I believe; however, the final remaining item is to clean up commits and commit messages. In particular:
* Each commit message should be descriptive of the change in a way that helps people in the future to know the purpose of the commit (see https://www.codelord.net/2015/03/16/bad-commit-messages-hall-of-shame/).
* Each commit message needs to be tagged with the same "GUACAMOLE-1733: " prefix at the front of it.

**Comment by ronansalmon** [comment]
ok, squashed the commits !

**Review by necouchman — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**